pkgs/development/tools/extism-cli/default.nix

```diff
@@ -1,32 +1,34 @@
-{ lib, stdenvNoCC, fetchFromGitHub, python3, makeBinaryWrapper }:
-
-stdenvNoCC.mkDerivation rec {
+{
+  pkgs ? import <nixpkgs> { system = builtins.currentSystem; }
+, lib ? pkgs.lib
+, buildGoModule ? pkgs.buildGoModule
+, fetchFromGitHub ? pkgs.fetchFromGitHub
+}:
+
+buildGoModule rec {
   pname = "extism-cli";
-  version = "0.1.0";
-
-  src = fetchFromGitHub {
-    owner = "extism";
-    repo = "cli";
-    rev = "97935786166e82154266b82410028482800e6061";
-    sha256 = "sha256-LRzXuZQt5h3exw43UXUwLVIhveYVFw/SQ2YtHI9ZnWc=";
-  };
-
-  buildInputs = [ python3 ];
-  nativeBuildInputs = [ makeBinaryWrapper ];
-
-  installPhase = ''
-    runHook preInstall
+  version = "0.3.8";
 
-    install -D -m 755 ./extism_cli/__init__.py "$out/bin/extism"
+  nativeBuildInputs = [ pkgs.makeBinaryWrapper ];
 
+  postFixup = ''
     # The extism cli tries by default to install a library and header into /usr/local which does not work on NixOS.
     # Pass a reasonable writable directory which can still be overwritten with another --prefix argument.
     wrapProgram "$out/bin/extism" \
       --add-flags '--prefix $HOME/.local'
-
-    runHook postInstall
   '';
 
+  src = fetchFromGitHub {
+    owner = "extism";
+    repo = "cli";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-StMipPMLSQzrhWv0yoKkNiuHMRW7QIhmVZ/M27WDWrM=";
+  } + /extism;
```

Comment:
- Can you please move the `src` after the `version`?
- What is intended with the last line? (` + /extism`)

Reply:
- Version has now been moved
- The `+ /extism` on the last line is because the tool exists in the `extism` directory instead of the root of the repo - is there a different way to handle this? I had looked into `subPackages` but ran into vendoring issues.

Reply:
I've updated this to use `modRoot` instead, as suggested below.